### PR TITLE
feat(engine): per-model chat-template override — Seed-OSS generation unblocked, promoted to Tested

### PR DESCRIPTION
**Pull request by magicnight**
## Summary

Resolves the Seed-OSS end-to-end blocker documented in #84: the checkpoint's `chat_template.jinja` uses integer dict keys, which swift-jinja 2.3.6 (latest; no upstream fix exists) cannot parse — template compilation threw before any token was generated.

**Mechanism** — per-model chat-template override, resolved at load in strict order: user file (`<model dir>/macmlx.chat_template.jinja`) → built-in override registered by `model_type` → the checkpoint's own template. Injected via swift-transformers' `ChatTemplateArgument.literal` at the `applyChatTemplate` seam; models with no override hit the exact pre-existing call path byte-for-byte. Applied overrides log their source; a present-but-unusable user file (empty / unreadable / non-UTF-8) logs a warning naming the reason — never silent in either direction.

**Built-in Seed-OSS template** — exactly ONE construct rewritten: the integer-keyed `budget_reflections_v05` dict (consumed as a dictsort threshold search) becomes an if/elif ladder emitting identical intervals. Everything else is verbatim upstream. Semantic equivalence is proven two ways:
- The capture script renders the ORIGINAL template (jinja2, cross-checked against `transformers.apply_chat_template`) into fixtures; a Swift test renders the OVERRIDE through swift-jinja and asserts full-string equality — 3 message scenarios + 21 thinking-budget boundary values (every tier edge, 0, negative, past-top), all byte-identical.
- The adversarial review independently re-rendered original-vs-override across 30 cases: 0 diffs.

**Headline** — the gated 19 GB real-weight smoke now PASSES on real Metal: 305 tokens at **18.6 tok/s**, with the model's native `<seed:think>` reasoning block engaging correctly (prompt validity proof). `docs/model-support.md` promotes Seed-OSS ⚠️ Parity-verified → **✅ Tested**.

Removal condition (documented in-code): drop the built-in override when swift-jinja gains integer object-key support. The user-file tier is a permanent power-user escape hatch for any model with a broken/incompatible template.

## Verification

- Adversarial review (opus): **APPROVE-WITH-NITS** — 0 CRITICAL/HIGH/MEDIUM. All four load-bearing claims independently verified: existential-dispatch seam correctness (explicit args provably match the non-override path's defaults), rewrite equivalence (30/30), swift-jinja version unified at 2.3.6 for test and runtime (SPM graph resolution), capture script honestly references the original (no self-reference trap). All 3 LOW findings fixed in-wave: unusable-user-file diagnostics + 4 tests (incl. invalid UTF-8 bytes), boundary parity re-anchored to original-template fixtures (hardcoded table demoted to secondary trip-wire), bare-environment fidelity caveat documented.
- Bare `swift test`: **503 tests / 62 suites, 0 failures** (XCTest subset 217, +21 new tests across the wave).
- `xcodebuild` full Metal: **TEST SUCCEEDED**, 503 all green.
- Real-weight smoke: passed on-device (evidence above); not re-run after the LOW fixes (test/fixture-only, load/generate path untouched).

---

> [!NOTE]
> **Medium Risk**
> Changes the shared inference tokenizer path for all models (nil override preserves prior behavior) and alters prompt tokenization for `seed_oss`; mitigated by regression tests and byte-for-byte parity fixtures.
> 
> **Overview**
> Adds **per-model chat-template overrides** at tokenizer load: optional `macmlx.chat_template.jinja` in the model directory, then built-ins keyed by `model_type`, otherwise the checkpoint template is unchanged. `HuggingFaceTokenizerLoader` resolves overrides, logs applied sources and warnings for unusable user files, and `TokenizerBridge` passes overrides to swift-transformers as `ChatTemplateArgument.literal` before Jinja compilation.
> 
> For **`seed_oss`**, registers `SeedOssChatTemplate`—a copy of the upstream template with only the integer-keyed thinking-budget dict replaced by an equivalent `if/elif` ladder so swift-jinja can parse it. Ungated **render-parity** tests (swift-jinja vs Python/jinja2 fixtures from `capture_seed_oss_chat_template.py`) and updated **Seed-OSS smoke** (no longer self-skips on the parser error) unblock end-to-end generation; **model-support** promotes Seed-OSS to Tested.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit fddf7538815760b9e7c4abc9bf2174e275c097ff. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: MacMLXCore/Sources/MacMLXCore/Engine/ChatTemplateOverride.swift [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        if fileManager.fileExists(atPath: userFile.path) {
+            if let data = try? Data(contentsOf: userFile) {
+                if let text = String(data: data, encoding: .utf8) {
+                    if !text.isEmpty {
+                        return Resolution(
+                            resolved: Resolved(
+                                template: text, source: "user file \(userOverrideFilename)"),
+                            skippedUserFileReason: nil)
+                    }
+                    skippedUserFileReason = "empty"
```
[the remark this answers is not in the document]
Comment (magicnight) [reply MacMLXCore/Sources/MacMLXCore/Engine/ChatTemplateOverride.swift:106]: Verified against main: confirmed — `!text.isEmpty` accepted a whitespace-only file as a live override, bypassing the "empty" diagnostic and rendering a blank prompt with no log. Fixed in #86: usability now checks trimmed emptiness, so a whitespace-only file takes the same "empty" skip path (loader warning + fallback) as a zero-byte file; regression test added.
